### DISCUSSION TITLE
[Question] GitHub URL of All Plugins Installed in a Vault

### DISCUSSION THREAD

**Discussion by FelipeRearden** [SilentVoid13/Templater#707] [Help] [answered]
Hello friends !!!!

Does anybody know if is possible to use Templater Script to get all GitHub links form **ALL** plugins that I have installed in a vault ?
![E452250B-145B-4FAC-B202-1E3021F7681F](https://user-images.githubusercontent.com/85180696/181245643-eff3c278-7fcd-4d92-baed-5dc95191f038.jpeg)

### My Idea

Since I generate a lot of FR and BR for the plugins beta plugins and plugins that I use daily, I want to create a workflow where I can transfer the content that I am writing in Obsidian directly to the repository of the plugin  on GitHub.

I already know how to do this workflow but Idont know how to automatically get all URL plugins from my Vault.

---

I hope somebody could give me some help 🙏

Thanks for reading this and have a great day :)

**Comment by welpdx** [comment, marked as answer]
What kind of output do you want?
Like this?
```
Templater: https://github.com/SilentVoid13
Editor Syntax Highlight: https://github.com/deathau
```

Also, what is BR and FR?

**Reply by AB1908** [reply]
Bug report and feature request presumably. I've also wanted a way to pull info like OS version, plugin version, stack trace etc automatically so we can prepopulate a bug report but haven't found the motivation to do it lol.

**Reply by FelipeRearden** [reply]
> Bug report and feature request presumably.

100% right @AB1908 !!!!!!

I've also wanted a way to pull info like OS version, plugin version, stack trace etc automatically so we can prepopulate a bug report but haven't found the motivation to do it lol.

That's a fantastic idea for BR's !!!!!!!

**Reply by welpdx** [reply]
OS version doesn't really change so you can just make that static.
Plugin version is doable.
Copying Stack Trace is much easier done [via console](https://developer.chrome.com/docs/devtools/javascript/reference/#:~:text=%23%20Copy%20stack%20trace,call%20stack%20to%20the%20clipboard). Kinda hard to do that with a script because it is hard to select what you want exactly. Might as well just copy it manually.

**Comment by welpdx** [comment]
This script outputs a list of installed plugins, with their name, version, and url:
```js
<%*
let plugins = Object.entries(this.app.plugins.plugins)
for (let i = 0; i < plugins.length; i++) {
let manifest = plugins[i][1]?.manifest
	
tR += manifest?.name + " v" + manifest?.version + ": " + manifest?.authorUrl + "\n"  ?? "no url?" + "\n" 
}
%>
```

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/QAchp4c.gif" alt="demo" />
</details>


Output:
```
Templater v1.12.0: https://github.com/SilentVoid13
Editor Syntax Highlight v0.1.3: https://github.com/deathau
```

**Reply by FelipeRearden** [reply]
This is absolutely amazing @welpdx !!!! 

Thank you very much for your help :)

Would you mind helping me how to prepend this information into a specific file ?

I use this script to overwrite the content of the note `InfoNotes.Drop.md`

```
<%*
const content = ""
const file = tp.file.find_tfile("InfoNotes.Drop")
await app.vault.modify(file,content)
%>
```

But I don't know how to modify your script to insert the content inside a specific note :(

Thank you very much 💙💙💙

**Reply by welpdx** [reply]
Just to clarify, you want the script output to be Prepended (put on the top) of the content in `InfoNotes.Drop.md`. Not replace or delete the content in the file but put in the beginning of the file right?

Also, do you have front matter in this `InfoNotes.Drop.md` file? Because if we need to add info at the beginning of this file, it would require the detection of frontmatter and putting it after frontmatter.

**Reply by FelipeRearden** [reply]
Forgive for not clarifying my workflow :)

> Just to clarify, you want the script output to be Prepended (put on the top) of the content in `InfoNotes.Drop.md`. Not replace or delete the content in the file but put in the beginning of the file right?

> Also, do you have front matter in this `InfoNotes.Drop.md` file? Because if we need to add info at the beginning of this file, it would require the detection of frontmatter and putting it after frontmatter.

I need overwrite the content of the note every time I run the Script.

The note does not have YAML. Thanks for your concern 💙

---

In my mediocre user mind I will do something like this to `InfoNotes.Drop.md`

1️⃣ Delete the Content of `InfoNotes.Drop.md` 

<%*
const content = ""
const file = tp.file.find_tfile("InfoNotes.Drop")
await app.vault.modify(file,content)
%>

2️⃣ Run your amazing script at `InfoNotes.Drop.md` from a different note (the note that I am writing the FR)

---

This way this Note will always stays updated when I have to run my workflow for creating a new FR/BR

I don't know how to do this in a one script, this way in my mind is easier to run the scripts at the same note :) 

---

Thanks for helping me :)

**Reply by welpdx** [reply]
try this:
```js
<%*
let content = ""  
const file = tp.file.find_tfile("InfoNotes.Drop")  

content += "Obsidian " + this.app.setting.settingTabs[4].containerEl.children[1].children[0].children[0]?.textContent

let plugins = Object.entries(this.app.plugins.plugins)
for (let i = 0; i < plugins.length; i++) {
let manifest = plugins[i][1]?.manifest
content += manifest?.name + " v" + manifest?.version + ": " + manifest?.authorUrl + "\n"  ?? "no url?" + "\n" 
}
await app.vault.modify(file,content)
%>
```

Edit: run script anywhere via `Alt + E` `Templater insert template modal`

**Reply by FelipeRearden** [reply]
Unfortunately I am getting this error after running the script with `ALT+E`

`Cannot read properties of undefined (reading 'children')`
![E0B26EB4-F232-458B-8CC6-259E91293F95](https://user-images.githubusercontent.com/85180696/181343495-fa5db308-2440-45e7-9967-42cc51dd1611.jpeg)

**Reply by welpdx** [reply]
Try this:
```js
<%*
let content = ""  
const file = tp.file.find_tfile("InfoNotes.Drop")  

content += "Obsidian v" + tp.obsidian.apiVersion + "\n"

let plugins = Object.entries(this.app.plugins.plugins)
for (let i = 0; i < plugins.length; i++) {
let manifest = plugins[i][1]?.manifest
content += manifest?.name + " v" + manifest?.version + ": " + manifest?.authorUrl + "\n"  ?? "no url?" + "\n" 
}
await app.vault.modify(file,content)
%>
```
I tried to add a line to read the obsidian version. It doesn't get generated if settings hasn't be opened. I've removed that line and replaced it with something else that works

**Reply by FelipeRearden** [reply]
Spectacular @welpdx !!!!!!!

Thank you very very very much for your help !!!!!!!!!!!!!!!!!!!

I am so happy now I finally can implement this workflow :)

Thanks again and I wish you a fantastic day !!!!!!!!! You are super 💙

**Reply by welpdx** [reply]
haha happy to help
